tests/ftqc/test_parametric_mid_measure.py

```diff
@@ -624,154 +623,127 @@ def test_diagonalize_mcm_transform_preserves_reset(self, reset):
-        diagonalized measurements that are stored in conditions on Conditional
+    def test_diagonalizing_mcm_used_as_cond(self):
+        """Test that when calling diagonalize_mcms, references to diagonalized
+        measurements that are stored in conditions on subsequent Conditional
         operators are updated to track the measurement on the tape following
         diagonalization, rather than the original object (for conditional
-        with MCM condition and MCM applied op)"""
+        with MCM condition and non-MCM op)"""
 
         with qml.queuing.AnnotatedQueue() as q:
             qml.RX(1.2, 0)
             mp = ParametricMidMeasureMP(0, angle=1.2, plane="YZ")
             mv = MeasurementValue([mp], processing_fn=lambda v: v)
-            qml.cond(mv == 0, partial(ParametricMidMeasureMP, angle=1.2))(wires=2, plane="XY")
+            # using qml.cond and conditionally applying a gate
+            qml.cond(mv == 0, partial(qml.RX, 1.2), partial(qml.RX, 2.4))(wires=2)
 
         original_tape = qml.tape.QuantumScript.from_queue(q)
         old_mp = original_tape.operations[1]
         assert isinstance(old_mp, ParametricMidMeasureMP)
 
+        # expected ops: RX, diagonalizing gate, new_mp, conditional RX, conditional RX
         (new_tape,), _ = diagonalize_mcms(original_tape)
-        assert len(new_tape.operations) == 6
-        for op in new_tape.operations[3:]:
-            assert isinstance(op, qml.ops.Conditional)
+        assert len(new_tape.operations) == 5
 
         new_mp = new_tape.operations[2]
         assert not isinstance(new_mp, ParametricMidMeasureMP)
         assert isinstance(new_mp, MidMeasureMP)
 
-        assert new_tape.operations[3].meas_val.measurements == [new_mp]
+        # the conditionals' MeasurementValues are mapped to the new, diagonalized mp
+        processing_fns = []
+        for op in new_tape.operations[3:]:
+            assert isinstance(op, qml.ops.Conditional)
+            assert op.meas_val.measurements == [new_mp]
+            processing_fns.append(op.meas_val.processing_fn)
+
+        # true and false processing fns are preserved on the conditionals
+        assert [fn(0) for fn in processing_fns] == [True, False]
+        assert [fn(1) for fn in processing_fns] == [False, True]
 
-    def test_diagonalizing_measurements_cond(self):
-        """Test that when calling diagonalize_mcms, references to previously
-        diagonalized measurements that are stored in conditions on Conditional
+    def test_diagonalizing_mcm_used_as_cond_and_op(self):
+        """Test that when calling diagonalize_mcms, references to diagonalized
+        measurements that are stored in conditions on subsequent Conditional
         operators are updated to track the measurement on the tape following
         diagonalization, rather than the original object (for conditional
-        with MCM condition and non-MCM op)"""
+        with MCM condition and MCM applied op)"""
```

Comment:
Fine for now, but I'm wondering if this explanation can be simplified a little

Reply:
Agreed - I wrote this a couple of weeks ago, and it already confused me when I looked at it again last week. Its not easy to read 🙈 I'll revisit briefly and see if I can improve this.

Reply:
I reworded these (this and the very similar one before it), I think its clearer now